clients/src/main/java/org/apache/kafka/clients/admin/DeleteTopicsResult.java

```diff
@@ -30,24 +31,56 @@
  */
 @InterfaceStability.Evolving
 public class DeleteTopicsResult {
-    final Map<String, KafkaFuture<Void>> futures;
+    private final Map<Uuid, KafkaFuture<Void>> topicIdFutures;
+    private final Map<String, KafkaFuture<Void>> nameFutures;
 
-    protected DeleteTopicsResult(Map<String, KafkaFuture<Void>> futures) {
-        this.futures = futures;
+    protected DeleteTopicsResult(Map<Uuid, KafkaFuture<Void>> topicIdFutures, Map<String, KafkaFuture<Void>> nameFutures) {
+        if (topicIdFutures != null && nameFutures != null)
+            throw new IllegalArgumentException("topicIdFutures and nameFutures can not both be specified.");
+        this.topicIdFutures = topicIdFutures;
+        this.nameFutures = nameFutures;
     }
 
+    protected static DeleteTopicsResult ofTopicIds(Map<Uuid, KafkaFuture<Void>> topicIdFutures) {
```

Comment:
Do these methods need to be protected or could they be package access?

Reply:
Forgive my ignorance but is that not the same thing? Is package private just that we can't use in subclasses outside the package? If that's the case then sure.

Reply:
Yes, that's right. The less exposed it is, the better.